### PR TITLE
Fix issue with valid PR title

### LINKED ISSUE #167: Check that the PR title starts with an imperative verb
https://github.com/FerretDB/github-actions/blob/6484839ddf046d22e9244c0447737a3275d8b7ad/conform-pr/main.go#L243-L251

That function should also check that title start with an imperative verb ("Fix …", not "Fixing …", "Fixed …" or other forms).

We could probably use https://github.com/jdkato/prose/tree/v2 (despite it being archived) and check that the first token has the correct POS tag.

### PR DESCRIPTION
**Pull request by adetunjii**
# Description
This fix introduces a hack around the issues with `prose` regarding wrong POS tags for some verbs.

Closes #167

* [x] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: conform-pr/main.go [hunk 1]
```diff
@@ -254,17 +254,25 @@ func checkTitle(_ *githubactions.Action, title string) error {
 		return fmt.Errorf("PR title must end with a latin letter or digit.")
 	}
 
-	doc, err := prose.NewDocument(strings.ToLower(title))
+	firstWordRegexp := regexp.MustCompile(`^\s*([a-zA-Z0-9]+)`)
+	groups := firstWordRegexp.FindStringSubmatch(title)
+
+	if groups == nil || len(groups) < 2 {
+		return fmt.Errorf("PR title must start with a latin letter or digit.")
```
Comment (AlekSi) [inline conform-pr/main.go:261]: We already check that is should start with an uppercase letter above

File: conform-pr/main.go [hunk 2]
```diff
@@ -254,17 +254,25 @@ func checkTitle(_ *githubactions.Action, title string) error {
 		return fmt.Errorf("PR title must end with a latin letter or digit.")
 	}
 
-	doc, err := prose.NewDocument(strings.ToLower(title))
+	firstWordRegexp := regexp.MustCompile(`^\s*([a-zA-Z0-9]+)`)
```
Comment (AlekSi) [inline conform-pr/main.go:257]: I think we could just use `strings.Split(title, " ")[0]`

File: conform-pr/main_test.go [hunk 3]
```diff
@@ -133,35 +133,54 @@ func TestCheckTitle(t *testing.T) {
...
-		title:       "test the title that does not start with an uppercase`",
-		expectedErr: errors.New("PR title must start with an uppercase letter."),
-	}}
+	}{
+		{
+			name:        "pull_request/title_without_dot",
+			title:       "Test the title without a dot",
+			expectedErr: nil,
+		},
+		{
```
Comment (AlekSi) [inline conform-pr/main_test.go:142]:
```suggestion
		}, {
```

Let's keep formatting as it was

File: conform-pr/main_test.go [hunk 4]
```diff
@@ -133,35 +133,54 @@ func TestCheckTitle(t *testing.T) {
...
+			title:       "test the title that does not start with an uppercase`",
+			expectedErr: errors.New("PR title must start with an uppercase letter."),
+		},
+		// edge cases where `prose` treats Nouns as verbs
+		{
+			name:        "pull_request/title_with_invalid_imperative_verb",
+			title:       "Please do not merge this PR",
+			expectedErr: nil,
+		},
+	}
```
Comment (AlekSi) [inline conform-pr/main_test.go:183]: Let's add actual words we encountered before: fix, document, bump

File: conform-pr/main_test.go [hunk 5]
```diff
@@ -133,35 +133,55 @@ func TestCheckTitle(t *testing.T) {
...
-		name:        "pull_request/title_with_backticks",
-		title:       "Test the title I'm a title with a `backticks`",
-		expectedErr: nil,
-	}, {
-		name:        "pull_request/title_without_uppercase",
-		title:       "test the title that does not start with an uppercase`",
-		expectedErr: errors.New("PR title must start with an uppercase letter."),
-	}}
+	}{
+		{
```
Comment (AlekSi) [inline conform-pr/main_test.go:137]:
```suggestion
	}{{
```

That will make `go fmt` happy

File: conform-pr/main_test.go [hunk 6]
```diff
@@ -133,35 +133,55 @@ func TestCheckTitle(t *testing.T) {
...
+			name:        "pull_request/title_with_invalid_imperative_verb",
+			title:       "Please do not merge this PR",
+			expectedErr: nil,
+		}, {
+			name:        "pull_request/title_with_invalid_imperative_verb",
+			title:       "A title without an imperative verb at the beginning",
+			expectedErr: nil,
+		},
+	}
```
Comment (AlekSi) [inline conform-pr/main_test.go:184]:
```suggestion
	}}
```

**Comment by adetunjii** [comment]
@AlekSi thanks for the updates. I'll take better care moving forward

File: conform-pr/main.go [hunk 7]
```diff
@@ -254,17 +254,20 @@ func checkTitle(_ *githubactions.Action, title string) error {
...
-	// TODO
-	_ = doc
-	// tag := doc.Tokens()[0].Tag
-	// if tag != "VB" {
-	// 	return fmt.Errorf("PR title must start with an imperative verb.")
-	// }
+	tokens := doc.Tokens()
+	tok := tokens[1]
+
+	// imperative verbs have either "VB" (?) or "VBP" tags
```
[the remark this answers is not in the document]
Comment (adetunjii) [reply conform-pr/main.go:266]: Yeah I decided to leave it in case of weird behavior since there's no way to predict what the tags would be

**Comment by adetunjii** [comment]
Removed the `VB` check
